src/ImageSharp/Colors/PackedPixel/Argb.cs

```diff
@@ -0,0 +1,309 @@
+        /// </param>
+        public Argb(Vector4 vector)
+        {
+            PackedValue = Pack(ref vector);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="packed">
+        /// The packed value.
+        /// </param>
+        public Argb(uint packed = 0)
+        {
+            PackedValue = packed;
+        }
+        /// <summary>
+        /// Gets or sets the red component.
+        /// </summary>
+        public byte R
+        {
+            get
+            {
+                return (byte)(PackedValue >> RedShift);
+            }
+
+            set
+            {
+                PackedValue = PackedValue & RedMask | (uint)value << RedShift;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the green component.
+        /// </summary>
+        public byte G
+        {
+            get
+            {
+                return (byte)(PackedValue >> GreenShift);
+            }
+
+            set
+            {
+                PackedValue = PackedValue & GreenMask | (uint)value << GreenShift;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the blue component.
+        /// </summary>
+        public byte B
+        {
+            get
+            {
+                return (byte)(PackedValue >> BlueShift);
+            }
+
+            set
+            {
+                PackedValue = PackedValue & BlueMask | (uint)value << BlueShift;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the alpha component.
+        /// </summary>
+        public byte A
+        {
+            get
+            {
+                return (byte)(PackedValue >> AlphaShift);
+            }
+
+            set
+            {
+                PackedValue = PackedValue & AlphaMask | (uint)value << AlphaShift;
+            }
+        }
+
+        /// <inheritdoc/>
+        public void PackFromVector4(Vector4 vector)
+        {
+            PackedValue = Pack(ref vector);
+        }
+
+        /// <inheritdoc/>
+        public Vector4 ToVector4()
+        {
+            return new Vector4(R, G, B, A) / MaxBytes;
+        }
+
+        /// <inheritdoc/>
+        public uint PackedValue { get; set; }
+
+        /// <inheritdoc/>
+        public void PackFromBytes(byte x, byte y, byte z, byte w)
+        {
+            PackedValue = Pack(x, y, z, w);
+        }
+
+        /// <summary>
+        /// Packs the four floats into a <see cref="uint"/>.
+        /// </summary>
+        /// <param name="x">The x-component</param>
+        /// <param name="y">The y-component</param>
+        /// <param name="z">The z-component</param>
+        /// <param name="w">The w-component</param>
+        /// <returns>The <see cref="uint"/></returns>
+        static uint Pack(float x, float y, float z, float w)
+        {
+            var value = new Vector4(x, y, z, w);
```

Comment:
@JimBobSquarePants this is super interesting, I've actually never run into these type problems due to `var`, thanks for sharing 😄 . I understand the perils now, up until now it was just the maintainability argument for me, but now I see how it could go wrong. 

> I honestly believe that it's having a negative effect on modern programming. Devs aren't learning their types nor the framework properly as a result and are being careless, using var mostly because they don't know what the return type of a method should be.

I fully agree here. I was fortunate enough to write a lot of C++ before coming to C# but I realize that might not be the norm these days.